x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -305,7 +305,7 @@ public void createApiKey(
             if (version.before(Authentication.VERSION_API_KEYS_WITH_REMOTE_INDICES) && hasRemoteIndices(request.getRoleDescriptors())) {
                 // Creating API keys with roles which define remote indices privileges is not allowed in a mixed cluster.
                 listener.onFailure(
-                    new IllegalStateException(
+                    new IllegalArgumentException(
```

Comment:
This change addresses [this comment](https://github.com/elastic/elasticsearch/pull/93414#discussion_r1119567913) (from Yang) in order to avoid failing with `500`.